### PR TITLE
use registry for creating the ids in t-digest lib

### PR DESCRIPTION
**Pull request by brharrington**
Updates the Json utility class to use an injected registry
to create the ids rather than using `DefaultId` directly.

This change does break backwards compatibility of some of
the classes to pass in the registry. However, it is still
experimental and our use of those classes is only via tests
or via a guice module so it will not break common usage.

cc @pstout

### CODE REVIEW

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java [hunk 1]
```diff
@@ -65,9 +69,10 @@ public StreamTDigestReader(DataInputStream in) {
    * Consume all data from the stream create a list of the measurement batches as they were
    * read from the stream.
    */
-  static List<List<TDigestMeasurement>> readAll(InputStream in) throws IOException {
+  static List<List<TDigestMeasurement>> readAll(Registry registry, InputStream in)
```
Comment (pstout) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java]: Should this method be made non-static and use the registry passed into the class constructor?  The constructor would need to be changed to keep a direct reference to the Registry and not just use it to construct the Json object.

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java [hunk 2]
```diff
@@ -27,17 +29,19 @@
...
 
+  private final Json json;
   private final DataInputStream in;
 
   private final byte[] buf = new byte[TDigestWriter.BUFFER_SIZE];
 
   /** Create a new instance. */
-  public StreamTDigestReader(InputStream in) {
-    this.in = new DataInputStream(in);
+  public StreamTDigestReader(Registry registry, InputStream in) {
```
Comment (pstout) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java]: Should this take a Json object rather than a Registry?

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/TDigestWriter.java [hunk 3]
```diff
@@ -43,6 +44,13 @@
   private final ByteBuffer buf = ByteBuffer.allocate(BUFFER_SIZE);
   private final ByteBufferOutputStream out = new ByteBufferOutputStream(buf, 2);
 
+  private final Json json;
+
+  /** Create a new instance. */
+  TDigestWriter(Registry registry) {
```
Comment (pstout) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/TDigestWriter.java]: Should this take a Json object rather than a Registry?

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java (hunk 1 above)
Comment (brharrington) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java]: I think it can change to just a `List<List<TDigestMeasurement>> readAll()` method that uses the stream. I would prefer to do that as a separate PR though.

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java (hunk 2 above)
Comment (brharrington) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java]: This class is public, the `Json` object is package private. I didn't want to expose that class outside the package.

File: spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java (hunk 1 above)
Comment (brharrington) [inline spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java]: Addressed in: https://github.com/Netflix/spectator/pull/183
